datafusion/core/src/physical_plan/hash_join.rs

```diff
@@ -1054,6 +1079,102 @@ fn equal_rows(
             DataType::LargeUtf8 => {
                 equal_rows_elem!(LargeStringArray, l, r, left, right, null_equals_null)
             }
+            DataType::Decimal(_, _) => {
```

Comment:
For decimal, I wonder if we also need to ensure that the precision and scale are the same (e.g. `l.data_type() == r.data_type()`) 🤔

Reply:
I think the scale should be the same, precision is insignificant.